src/simplewallet/simplewallet.cpp

```diff
@@ -4760,10 +4760,11 @@ bool simple_wallet::try_connect_to_daemon(bool silent, uint32_t* version)
   uint32_t version_ = 0;
   if (!version)
     version = &version_;
-  if (!m_wallet->check_connection(version))
+  expect<void> status{};
+  if (!(status = m_wallet->check_connection(version)))
```

Comment:
That is *really* confusing. Above, you inverted the return value check when swithing from bool to expect. Here and below, not.

Reply:
No longer using `expect<void>`, everything is `boost::system::error_code` with the expected returns.